src/main/java/com/shima/todo_list/TodoListController.java

```diff
@@ -15,9 +22,30 @@ public TodoListController(TodoListService todoListService) {
         this.todoListService = todoListService;
     }
 
-    @GetMapping("/todo_Lists")
+    //GET 指定したIDが存在しない場合
+    @ExceptionHandler(value = UserNotFoundException.class)
+    public ResponseEntity<Map<String, String>> handTaskNotFoundException(
+            UserNotFoundException e, HttpServletRequest request) {
+        Map<String, String> body = Map.of(
+                "timestamp", ZonedDateTime.now().toString(),
+                "status", String.valueOf(HttpStatus.NOT_FOUND.value()),
+                "error", HttpStatus.NOT_FOUND.getReasonPhrase(),
+                "message", e.getMessage(),
+                "path", request.getRequestURI());
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
+    }//404エラーを返す
+
+
+    @GetMapping("/todo_lists")
     public List<TodoList> getTodoList() {
         List<TodoList> todoLists = todoListService.getTodoList();
         return todoLists;
     }
+
+    //GET
+    //idで指定したデータ取得と例外処理
+    @GetMapping("/todo_lists/{id}")
+    public TodoList getTodoList(@PathVariable("id") int id) throws UserNotFoundException {
```

Comment:
UserNotFoundExceptionはRuntimeExceptionを継承しているので throws は不要なはずです。
```suggestion
    public TodoList getTodoList(@PathVariable("id") int id) {
```

Reply:
削除しました。

Reply:
OKです。
コミットハッシュをはってくれるとありがたい。